torchcam/cams/cam.py

```diff
@@ -422,18 +427,18 @@ def _get_weights(self, class_idx, scores=None):
 
         # Disable hook updates
         self._hooks_enabled = False
+        fmap = 0
 
         for _idx in range(self.num_samples):
-            noisy_m = (_idx + 1) / self.num_samples * self._input * (upsampled_a +
-                                     self._distrib.sample(self._input.size()).to(device=self._input.device))
+            fmap += (_idx + 1) / self.num_samples * self._input * upsampled_a
```

Comment:
This should probably be `fmap += ((_idx + 1) / self.num_samples) * self._input * upsampled_a`
Not sure if this is the execution that's carried out :-)